src/main/java/com/saralein/server/server/SetupServer.java

```diff
@@ -0,0 +1,36 @@
+package com.saralein.server.server;
+
+import com.saralein.server.ShutdownHook;
+import com.saralein.server.connection.ListeningSocket;
+import com.saralein.server.connection.ServerSocket;
+import com.saralein.server.logger.Logger;
+import com.saralein.server.logger.ConnectionLogger;
+import com.saralein.server.response.Response;
+import com.saralein.server.response.ResponseBuilder;
+
+import java.io.IOException;
+
+public class SetupServer {
+    private static int getPort(String[] args) {
+        return Integer.parseInt(args[0]);
+    }
+
+    public static Server setup(String[] args) {
+        Integer port = getPort(args);
+        Logger logger = new ConnectionLogger(System.out);
+        ServerSocket serverSocket = null;
+        Response responseBuilder = new ResponseBuilder();
+
+        try {
+            serverSocket = new ListeningSocket(port);
+        } catch (IOException e) {
+            logger.log(e.getMessage());
+        }
+
+        Server server = new Server(serverSocket, logger, responseBuilder);
+
+        Runtime.getRuntime().addShutdownHook(new ShutdownHook(server, logger));
```

Comment:
What, no way, I've never seen this before. Very Cool. Is there a way we can inject a runtime instead of calling it statically?

Reply:
OK, I [injected runtime](https://github.com/saralein/java-server/pull/1/commits/19cabc629184fea0b9134629339694b3d3a5e4c2#diff-5d0e8a54ffcbf82852ec9a9b532910f3R13) into `SetupServer`.